providers/upgrade-zfsLocalPV/test.yml

```diff
@@ -0,0 +1,209 @@
+- hosts: localhost
+  connection: local
+    
+  vars_files:
+  - test_vars.yml
+    
+  tasks:
+
+  - block:
+    
+    ## Generating the testname for deployment
+     - include_tasks: /utils/fcm/create_testname.yml
+    
+    ## RECORD START-OF-TEST IN LITMUS RESULT CR
+     - include_tasks: /utils/fcm/update_litmus_result_resource.yml
+       vars:
+         status: 'SOT'
+
+     - name: Verify that the ZFS-controller pod and zfs-node daemonset all are in running state
+       shell: >
+         kubectl get pods -n kube-system -l role=openebs-zfs
+         --no-headers -o custom-columns=:status.phase | sort | uniq
+       args: 
+         executable: /bin/bash
+       register: ZFS_driver_components
+       failed_when: "'Running' not in ZFS_driver_components.stdout"
+
+     - name: Get the version tag for zfs-driver
+       shell: >
+         kubectl get sts openebs-zfs-controller -n kube-system 
+         -o jsonpath='{.spec.template.spec.containers[?(@.name=="openebs-zfs-plugin")].image}' | cut -d ":" -f2
+       args:
+         executable: /bin/bash
+       register: zfs_driver_tag
+
+     - name: Get the replica count for zfs-controller statefulset
+       shell: >
+         kubectl get sts openebs-zfs-controller -n kube-system -o jsonpath='{.status.replicas}'
+       args: 
+         executable: /bin/bash
+       register: no_of_zfs_ctrl_replicas
+
+     - name: Record the number of zfs-controller replicas
+       set_fact:
+         zfs_ctrl_replicas: "{{ no_of_zfs_ctrl_replicas.stdout }}"
+
+     - name: Get the list of node-agent pods in openebs-zfs-node daemonset
+       shell: >
+         kubectl get po -n kube-system -l app=openebs-zfs-node --no-headers -o custom-columns=:.metadata.name
+       args:
+         executable: /bin/bash
+       register: ds_pods
+         
+     - block:
+       ## This task creates new CRDs as zfs-LocalPV related CRs are now grouped 
+       ## under `zfs.openebs.io` from v0.6 release.
+       - name: Apply the new CRDs for zfs-LocalPV 
+         shell: >
+           kubectl apply -f https://raw.githubusercontent.com/openebs/zfs-localpv/master/upgrade/crd.yaml
+         args:
+           executable: /bin/bash
+         register: new_crds
+         failed_when: "new_crds.rc != 0"
+
+       ## This task create new CRs for zfs-volume and zfs-snapshot with updated
+       ## apiversion to `zfs.openebs.io`. Previously this was `openebs.io`.
+       - name: Download the Upgrade script for creating new CRs with apiversion as `zfs.openebs.io`
+         get_url:  
+           url: https://raw.githubusercontent.com/openebs/zfs-localpv/master/upgrade/upgrade.sh
```

Comment:
from master?

Reply:
This upgrade.sh file is creating new CRs with upgraded apiversion only. and these steps are required only when upgrading from v0.4 and v0.5.
So taking these files from master is good.